pkg/okteto/client.go

```diff
@@ -85,6 +85,30 @@ func query(ctx context.Context, query string, result interface{}) error {
 	return nil
 }
 
+func queryWithVariables(ctx context.Context, query string, variables map[string]interface{}, result interface{}) error {
```

Comment:
could we add a test for this? not end to end, but just to make sure we are sending the right thing to the graphql endpoint? I'm concerned that, since this could be any value, that we might need to escape things.

Reply:
Should we also check for command injection, since this is a free-formed value?

Reply:
We could, but we are delegating that logic into the third party library [`machinebox/graphql`](https://github.com/machinebox/graphql), so adding tests for that would mean to add tests to their library.

It would probably need to be an end to end test (maybe with fake API server like they are doing on their tests), because the library receives a Query object and not the actual string representation that will be sent to the graphql endpoint.

Reply:
At the end, I've changed this function. Since variables are defined in the query string, it makes sense that variables are set in the same place where the query is defined. In this case, `pipeline.go`. The idea is that all variables should be set using their types instead of using free-formed types like `interface{}`